internal/provider/waypoint/resource_waypoint_template.go

```diff
@@ -60,6 +61,13 @@ type tfcWorkspace struct {
 	TerraformProjectID types.String `tfsdk:"terraform_project_id"`
 }
 
+func (t tfcWorkspace) attrTypes() map[string]attr.Type {
+	return map[string]attr.Type{
+		"name":                 types.StringType,
+		"terraform_project_id": types.StringType,
+	}
+}
```

Comment:
It's not clear to me what this is fixing or doing that we weren't doing before, since the field is still a `SingleNestedAttribute`.. what improvements does this make?

Reply:
That is there so that you can call `tfcWorkspace.attrTypes()` in places like line 337 of resource_waypoint_add_on_definition.go